### PR TITLE
Fixing errors using endpoint/CLI for conversion/validation

### PR DESCRIPTION
**Pull request by abayer**
We didn't have @Library in our classloader, so we'd get errors. That
was annoying. So here is a fix.

@kzantow - can you do a local build of this and verify it works right for you now?

cc @reviewbybees esp @kzantow @rsandell

### CODE REVIEW

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy [hunk 1]
```diff
@@ -106,12 +106,18 @@ public class Converter {
...
             new CodeSource(src, new Certificate[0]),
-            new GroovyClassLoader());
+            getCompilationClassLoader());
         cu.addSource(src);
 
         return compilationUnitToPipelineDef(cu)
     }
 
+    private static GroovyClassLoader getCompilationClassLoader() {
+        Jenkins j = Jenkins.instance
```
Comment (kzantow) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy]: y not the `@Nonnull` `Jenkins.getInstance()` ??

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy [hunk 2]
```diff
@@ -200,7 +211,7 @@ public class Converter {
 
         String rawScript = execution.script
 
-        return scriptToPipelineDef(rawScript)
+        return scriptToPipelineDef(rawScript, CpsThread.current()?.getExecution()?.getShell()?.classLoader)
```
Comment (kzantow) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy]: Why not put this in the get `getCompilationClassLoader` and use that here?

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy (hunk 1 above)
Comment (abayer) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy]: Durrr. Good point. `Jenkins.instance` is `Jenkins.getInstance()` (thanks, Groovy magic!), but there's no need for the null check.

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy (hunk 2 above)
Comment (abayer) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy]: Smart man.

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy [hunk 3]
```diff
@@ -106,23 +107,33 @@ public class Converter {
...
 
         return compilationUnitToPipelineDef(cu)
     }
 
+    private static GroovyClassLoader getCompilationClassLoader(GroovyClassLoader inputLoader = null) {
+        if (inputLoader != null) {
+            return inputLoader
+        } else {
+            Jenkins j = Jenkins.instance
+            ClassLoader cl = j != null ? j.getPluginManager().uberClassLoader : new GroovyClassLoader()
```
Comment (kzantow) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy]: `@Nonnull Jenkins.getInstance()` ?

**Comment by abayer** [comment]
@kzantow All of them addressed in latest commit!

**Comment by rsandell** [comment]
Injecting uberClassaLoader like that might override the sandbox.

**Comment by abayer** [comment]
This isn't in the Pipeline code - it's all sandbox-free. =)

**Comment by abayer** [comment]
Plus the only way we ever end up hitting *that* code path is if we're not even being called within a `CpsThread`.

**Comment by jglick** [comment]
Recommend adding some coverage to `acceptance-test-harness`. Would suffice to have one end-to-end test that pulls in a library and a bunch of tricky features and proves everything works.

**Comment by abayer** [comment]
@jglick Yeah, I need to revisit my ATH PR.

**Pull request merged**